pkg/grafana/client.go

```diff
@@ -158,6 +146,14 @@ func (c *Client) UpdateOrgUserAuthz(user gapi.User, groups config.Groups) (userO
 		}
 	}
 
+	// only update global admin value if it's different to what the user already have
+	if user.IsAdmin != isGlobalAdmin {
```

Comment:
Did you find a way to consistently reproduce this bug? Was the issue due to sort order before? We don't ever get an updated use object from Grafana so we don't have a great way to know if this worked. The only way you could assert that this fixes the problem is to see if c.UpdateUserPermissions is called with a value you don't expect or when you don't expect it to be. That could be hard if this was based on sort order because it would fail X/Y% of the time where X is the number of groups claims with isGlobalAdmin true and Y is the total number of groups claims.

Reply:
Couldn't find a way to reproduce it. In any case applying only one instead of per group is better I think.
In general testing this is difficult for what you mentioned, the user is updated on the server so don't have a way to assert if the `isAdmin` value changed. I updated the mock at some point to explicitly return an error on `UpdateUserPermissions` when the user "is not valid" by forcing id 0